File: Problemas/2-ejercicio/Problemas_simple_lineal.py
# Importar librerías necesarias
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# Leer datos desde un archivo CSV
dataset = pd.read_csv('Salario_Data.csv')

# Dividir los datos en características independientes (X) y variable dependiente (Y)
x = dataset.iloc[:, :-1].values  # Todas las columnas excepto la última
y = dataset.iloc[:, -1].values   # Solo la última columna

# Dividir los datos en conjuntos de entrenamiento y prueba
x_Capacitar, x_Prueba, y_Capacitar, y_Prueba = train_test_split(
    x, y, test_size=1/3, random_state=0
)

# Crear y entrenar el modelo de regresión lineal
regresar = LinearRegression()
regresar.fit(x_Capacitar, y_Capacitar)  # Entrenar el modelo con datos de entrenamiento

# Realizar predicciones en el conjunto de prueba
y_prediccion = regresar.predict(x_Capacitar)


# visualizar los datos de entranamiento
plt.scatter(x_Capacitar, y_Capacitar, color="blue")
plt.plot(x_Prueba, regresar.predict(x_Prueba), color="black")
plt.title("Sueldo por anios de experiencia")
plt.xlabel("Anios de experiencia")
plt.ylabel("sueldo base al trabajo")
plt.show()

# visualizar los datos de pruebas o entrenamiento
plt.scatter(x_Prueba, y_Prueba, color="purple")
plt.plot(x_Prueba, regresar.predict(x_Prueba), color="black")
plt.title("Sueldo por anios de experiencia")
plt.xlabel("Anios de experiencia")
plt.ylabel("sueldo base al trabajo")
plt.show()

# aprendizaje automatizado Para Python y los resultados son mayor mente sactisfactorio

File: Problemas/2-ejercicio/Problemas_simple_lineal.R
# Regresion Lineal Simple

dataset = read.csv('C:/Users/david/Music/MachingLearning/Problemas/2-ejercicio/Salario_Data.csv')


library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
# Conjunto de entrenamiento (datos donde split es TRUE)
entrenamiento_enviar = subset(dataset, split == TRUE)

# Conjunto de prueba (datos donde split es FALSE)
prueba_enviar = subset(dataset, split == FALSE)


# escalado de valores
# entrenamiento_enviar[,2:3] = scale(entrenamiento_enviar[,2:3])
# prueba_enviar[,2:3] = scale(prueba_enviar[,2:3])

# Ajustar el modelo de regresionj lineal simple con el conjunto de entrenaimiento
regresar = lm(formula = Salary ~ YearsExperience, data = entrenamiento_enviar) #se tiene que parecer las variables del dataset = YearsExperience

# Predicir resultados con el conjunto de prueba
y_prediccion = predict(regresar, newdata = prueba_enviar)

# Visualizar resultados del conjunto de entrenamiento
ggplot(data = entrenamiento_enviar, aes(x = YearsExperience)) +
  # Puntos de los datos reales
  geom_point(aes(y = Salary), color = "blue") +
  # Línea del modelo de regresión
  geom_line(aes(y = predict(regresar, newdata = entrenamiento_enviar)), color = "green") +
  # Etiquetas y título
  ggtitle("sueldo vs años de experiencia (conjuento de entrenamiento)") +
  xlab("Años de experiencia")+
  ylab("sueldo (en $)")

# Visualizar resultados del conjunto de prueba
ggplot(data = prueba_enviar, aes(x = YearsExperience)) +
  # Puntos de los datos reales
  geom_point(aes(y = Salary), color = "blue") +
  # Línea del modelo de regresión
  geom_line(aes(y = predict(regresar, newdata = prueba_enviar)), color = "green") +
  # Etiquetas y título
  ggtitle("sueldo vs años de experiencia (conjuento de prueba)") +
  xlab("Años de experiencia")+
  ylab("sueldo (en $)")




